private-templates-service/client/src/store/currentTemplate/actions.ts

```diff
@@ -193,6 +193,15 @@ export function updateTemplate(templateID?: string, currentVersion?: string, tem
     else {
       newTemplate.template = appState.currentTemplate.templateJSON;
     }
+
+    if (sampleDataJSON) {
+      // TODO: the clientSDK, backend, and DB adapter should be refactored such that the data is a json object and not an array of objects
+      newTemplate.data = new Array(sampleDataJSON);
+    }
+    else {
+      newTemplate.template = appState.currentTemplate.sampleDataJSON;
```

Comment:
should this be 
`newTemplate.data = appState.currentTemplate.sampleDataJSON;`

Reply:
My bad, that was a mistake. I've updated the logic. Could you please take another look?